integrations/git_helper_for_declarative_test.go

```diff
@@ -139,6 +139,9 @@ func doGitInitTestRepository(dstPath string) func(*testing.T) {
 	return func(t *testing.T) {
 		// Init repository in dstPath
 		assert.NoError(t, git.InitRepository(dstPath, false))
+		repo, err := git.OpenRepository(dstPath)
+		assert.NoError(t, err)
+		assert.NoError(t, repo.SetDefaultBranch("master"))
```

Comment:
We need `repo.Close()` here.

Reply:
Or to make the test run fast, can we check the default name before test? Then we do not need `SetDefaultBranch("master")` again and again.

Reply:
I added the repo.Close(), but shouldn't SetDefaultBranch be checking already if there is a need to change the branch rather that doing it in the test ?

Reply:
So since GetDefaultBranch would run git symbolic-ref HEAD, and SetDefaultBranch would run git symbolic-ref HEAD refs/head/foo, my intuition is that checking is as fast as changing directly. For N tests being run for 1 repo, you have either N time SetDefaultBranch, or N run of GetDefaultBranch + 1 run of SetDefaultBranch. That's N+1 run of git vs N run of git.

But that's assuming that the biggest part of the slowdown is due to running a external process, and that the overhead of SetDefaultBranch vs GetDefaultBranch is very small. I am far from being a git specialist, but reading the code, it seems that setting the default branch resolve the branch name, then acquire a lock, update the file, and that's it. 

While I can't claim that my testing is rigorous, running both commands with time  yield similar execution time around 4 to 10 milliseconds, and none looked faster than the other, so I suspect the execution time is mostly dominated by the start of git as a binary rather than the code path taken (for theses 2 commands).